dissect/target/plugin.py

```diff
@@ -793,8 +797,154 @@ def _modulepath(cls) -> str:
+        for entry in self.SUBPLUGINS:
+            try:
+                subplugin = getattr(self.target, entry)
+                self._subplugins.append(subplugin)
+            except Exception:  # noqa
+                target.log.exception("Failed to load subplugin: %s", entry)
+
+    def _func(self, func_name: str, subplugins: list) -> Iterator[Record]:
+        """Return the supported subplugin records.
+
+        Args:
+            func_name: Exported function of the subplugin to find.
+
+        Yields:
+            Record from the sub function.
+        """
+        for entry in subplugins:
+            try:
+                subplugin = getattr(self.target, entry)
+                for item in getattr(subplugin, func_name)():
+                    yield item
+            except Exception:
+                continue
+
+    def check_compatible(self) -> None:
+        if not len(self._subplugins):
+            raise UnsupportedPluginError("No compatible subplugins found")
+
+    def __init_subclass_namespace__(cls, **kwargs):
+        # If this is a direct subclass of a Namespace Plugin,
+        # create a reference to the current class for indirect subclasses
+        # so that the can autogenerate aggregate methods there
+        cls.__nsplugin__ = cls
+
+    def __init_subclass_subplugin__(cls, **kwargs):
+        # Does the direct subclass already have a SUBPLUGINS attribute?
+        # if not, create this attribute on the direct subclass
+        if not getattr(cls.__nsplugin__, "SUBPLUGINS", None):
+            cls.__nsplugin__.SUBPLUGINS = set()
+
+        # Register the current plugin class as a subplugin with
+        # the direct subclass of NamespacePlugin
+        cls.__nsplugin__.SUBPLUGINS.add(cls.__namespace__)
+
+        # Collect the public attrs of the subplugin
+        for subplugin_func_name in cls.__exports__:
+            subplugin_func = inspect.getattr_static(cls, subplugin_func_name)
+
+            # The attr need to be callable and exported
+            if not isinstance(subplugin_func, Callable):
+                continue
+
+            # The method needs to have a single record descriptor as output
+            if not isinstance(getattr(subplugin_func, "__record__", None), RecordDescriptor):
+                continue
+
+            # The method needs to be part of the current subclass and not a parent
+            if not subplugin_func.__qualname__.startswith(cls.__name__):
+                continue
+
+            # If we already have an aggregate method, skip
+            if existing_aggregator := getattr(cls.__nsplugin__, subplugin_func_name, None):
+                existing_aggregator.__subplugins__.append(cls.__namespace__)
+                continue
+
+            # The generic template for the aggregator method
+            def generate_aggregator(method_name):
+                def aggregator(self) -> Iterator[Record]:
+                    yield from self._func(method_name, aggregator.__subplugins__)
```

Comment:
Perhaps inline the `_func` logic here? I'm not sure I see the benefit anymore of having that separate.